src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -1467,6 +1468,12 @@ private void VisitQueryableMethodCall(MethodCallExpression expression)
                         var type = expression.Arguments[0].Type.GetGenericArguments()[0];
                         _documentQuery.AddRootType(type);
                     }
+
+                    if (expression.Type.GetTypeInfo().IsGenericType)
+                    {
+                        _ofType = expression.Type.GetGenericArguments()[0];
```

Comment:
What happens when `OfType` is used after the Select?

Reply:
I think this implementation works because `AddToFieldsToFetch` is called when we are inside `Select` processing. Although let's add a test for that (to ensure `OfType` used after the `Select` doesn't break anything). Also it might be good idea to skip setting `_ofType` field if  `Select` was already processed

Reply:
a query with `OfType` after `select`, like this one:
```
var query = session.Query<Result, UsersIndex>()
	.Select(u => u.Friend)  //Friend type is 'User'
	.OfType<Result>();
```
will still be processing `OfType` _before_ processing the `Select`
so the `_ofType` variable would be set before we reach `AddToFieldToFetch`, which seems a bit wrong.. 
but it will only matter if the select is selecting Id, and I'm not sure that this is a valid use-case.
thoughts?

Reply:
Do we have any way to figure out if `OfType` was before or after `Select`? We need to use the type specified by `OfType` only if it precedes the projection